### ISSUE TITLE
exception message should also appear in repr (not only in response)

### ISSUE THREAD

**Issue by shackenberg** [pallets/werkzeug#728]
Hi, 

I searched the issues but could not find anything related. 

I want to change the description of an error, like shown in the [last line on that docu page about exceptions](http://werkzeug.pocoo.org/docs/0.10/exceptions/#custom-errors). 

When I enter
 ```
python -c "import werkzeug; print werkzeug.__version__; raise werkzeug.exceptions.HTTPException('Can you read me?')"
```
the output is:
```
0.10.4
Traceback (most recent call last):
  File "<string>", line 1, in <module>
werkzeug.exceptions.HTTPException
```

But I would expect something like:
```
...
werkzeug.exceptions.HTTPException: Can you read me?
```

I also tried it with the newest version on github, but I get the same output:
```
0.11.dev0
Traceback (most recent call last):
  File "<string>", line 1, in <module>
werkzeug.exceptions.HTTPException
```

Python version is
```
$ python -V
Python 2.7.6
```

**Comment by untitaker** [comment]
Your custom error message is shown in the HTTP response. I agree that it'd be nice to show it in the repr as well.

**Comment by Aayush-Kasurde** [comment]
I would like to work on this enhancement. Can anyone help me in working on this ?

**Comment by untitaker** [comment]
@Aayush-Kasurde I'm not sure what you need help with (please elaborate), relevant code is at: https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/exceptions.py#L162, tests at https://github.com/mitsuhiko/werkzeug/blob/master/tests/test_exceptions.py

If you're new to GitHub, this seems like a decent guide for submitting PRs: https://guides.github.com/activities/contributing-to-open-source/

**Comment by Aayush-Kasurde** [comment]
@untitaker Thanks for Guide. 

I am new to this project so I need help in understanding - what is expected in exception? 

I will work and let you know using PR.

**Comment by untitaker** [comment]
>what is expected in exception?

It's expected that the string returned by the `__repr__` method (linked above) somehow includes `self.description` for debugging purposes.

**Comment by shellhead** [comment]
Since there hasn't been any conversation on this for a year, I'll be glad to pick this up!

**Comment by untitaker** [comment]
Okay, go for it! Make sure to mention the PR here then.

**Comment by shellhead** [comment]
I'm unable to recreate the original issue. When I run the input provided, I get back
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
werkzeug.exceptions.HTTPException: <exception str() failed>
```
Seems to be failing when calling `__str__` as the `code` is `None`, causing problems on the string formatting.

Which leads me to believe we actually want to make changes in `__str__` rather than `__repr__`. I've made a PR with that in mind. Please let me know if that's acceptable or if I missed something.

**Comment by shellhead** [comment]
@untitaker, any suggestions or advice on this?

**Comment by jeffwidman** [comment]
I think this is fixed by #970, just not auto-closed by github

**Comment by untitaker** [comment]
Thanks again @shellhead!
